platform/core.multitabs/src/org/netbeans/core/multitabs/impl/DocumentSwitcherTable.java

```diff
@@ -174,7 +177,27 @@ public String getToolTipText( MouseEvent event ) {
         }
         return null;
     }
-
+    
+    boolean closeSelectedDocumentList() {
+        List<TabData> tabs = controller.getTabModel().getTabs();
+        Item item = ( Item ) getModel().getValueAt( getSelectedRow(), getSelectedColumn());
+        ProjectProxy project = item.getProject();
+        ProjectSupport projectSupport = ProjectSupport.getDefault();
+        int numOfOtherTabs = 0;
+        for ( int i = tabs.size(); i-- > 0; ) {  //reverse iteration to close tabs from the end so tabIndex does not change
```

Comment:
Probably better to assemble tabs to be closed in a list first, then iterate over them normally. There may be "Save Changes?" dialogs popping up as a result of trying to close the tabs, and they normally come left-to-right.

Reply:
Also, there's an off-by-one error here. Reverse for loops over zero-based indices should go "for (int i = N - 1; i >= 0; i--)". (But better iterate left-to-right over a copy of the relevant tabs as suggested above.)

Reply:
I am sorry, but I dont think I understand the first suggestion.

Reply:
a defensive copy would look like:
`List<TabData> tabs = new ArrayList<>(controller.getTabModel().getTabs());`
instead of
`List<TabData> tabs = controller.getTabModel().getTabs();`

now you could simply iterate through `tabs` with a normal for loop since it is a copy of the original and won't change while you iterate.